pkg/builtin/unary/asin.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unary
+
+import (
+	"fmt"
+
+	"github.com/matrixorigin/matrixone/pkg/builtin"
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend/overload"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/asin"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func init() {
+	extend.FunctionRegistry["asin"] = builtin.Asin
```

Comment:
Move all implementation about asin, acos to the sql/plan2/function/builtin/unary. The path pkg/builtin/unary may be replaced.

Reply:
In this pr, I have implemented `asin` both in `pkg/builtin/unary` and `pkg/sql/plan2/function/builtin/unary`. So does it mean that I just delete the implementation in `pkg/builtin/unary`, and reserve the implementation in `pkg/sql/plan2/function/builtin/unary`?

Reply:
yes